### ISSUE TITLE
Crash while handling Not Found response

### ISSUE THREAD

**Issue by irnc** [terraform-providers/terraform-provider-azurerm#766]
### Terraform Version

Terraform v0.11.2
+ provider.azurerm v1.1.0

### Affected Resource(s)

It seams that all resources are affected, as all could receive Not Found response.

### Terraform Configuration Files

```
data "azurerm_app_service_plan" "ServicePlanb" {
  name = "ServicePlan"
  resource_group_name = "A"
  # while ServicePlan actually exists under resource group B
}
```

### Panic Output

https://gist.github.com/irnc/0b84ce980d6af2c5861d7499c38b12b8 contains only extract with panic stack trace.

### Expected Behavior

Not Found error should be handled.

### Actual Behavior

Terraform crashes on a code branch, which was not supposed to be run, i.e. `err != nil` branch was expected to be run instead of https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.1.0/azurerm/data_source_app_service_plan.go#L97

### Steps to Reproduce

- `terraform plan`

**Comment by sjwaight** [comment]
Recreate this scenario by deleting the Resource Group containing your previously provisioned resources and you will crash out of Terraform as per the above.

**Comment by tombuildsstuff** [comment]
hey @irnc @sjwaight 

Thanks for opening this issue (and for the reproduction steps!)

I've taken a look into this issue and have opened #849 which contains a fix - which should be going out in v1.1.2 later today :)

Thanks!
